pkg/sources/forward/data_forward.go

```diff
@@ -192,6 +192,31 @@ type readWriteMessagePair struct {
 	transformerError error
 }
 
+func processReadMessages(hasUDTransformer bool, readMessages []*isb.ReadMessage, processorWM wmb.Watermark,
```

Comment:
Please write a small paragraph as a comment about what this function does, especially what it **owns**. If it owns processing the messages from the ReadMessages list and generating WriteMessages list ready for writing to buffer, then strictly there should be no concept of UDTransformer outside of this function.